libbeat/scripts/kibana_extract.py

```diff
@@ -0,0 +1,144 @@
+
+import os
+import json
+import jinja2
+
+
+def unescape_json(escaped):
+    obj = json.loads(escaped)
+    return json.dumps(obj, sort_keys=True, indent=4, separators=(',', ': '))
+
+def load_json(filepath):
+    json_data=open(filepath).read()
+    return json.loads(json_data)
+
+def load_search_source(file):
+    data = load_json(file)
+
+    field = "searchSourceJSON"
+
+    if data.has_key("kibanaSavedObjectMeta") and data["kibanaSavedObjectMeta"].has_key(field):
+        jsonData = unescape_json(data["kibanaSavedObjectMeta"][field])
+    else:
+        return ""
+
+    write_output(file, field, jsonData)
+
+
+def load_field(file, field):
+    data = load_json(file)
+
+    if  data.has_key(field):
+        jsonData = unescape_json(data[field])
+    else:
+        return ""
+
+    write_output(file, field, jsonData)
+
+
+def write_output(file, field, jsonData):
+
+    dir_name = os.path.splitext(file)[0]
+
+    if not os.path.exists(dir_name):
+        os.mkdir(dir_name)
+
+    with open(dir_name + "/" + field + ".json", 'w') as outfile:
+        outfile.write(jsonData)
+
+
+def extract():
+    base = './etc/kibana'
+
+    folders = os.listdir(base)
```

Comment:
Lots of good information in here:
* Dashboards: It seems like we need a naming convention for dashboards. Metricbeat will have more then one dashboard per beat. Would the convention be `beatname module`. A dashboard must have a unique name, correct?
* Visualizations: The same seems to be true for visualization. We will need a unique name across all beats and should also make sure to now have generic names for visualizations, as otherwise they could overwrite existing visualizations. Is it required to use the name as id?
* Will different beats ever share searches / visualisations?